lasagne/updates.py

```diff
@@ -262,10 +251,52 @@ def nesterov_momentum(loss_or_grads, params, learning_rate, momentum=0.9):
 
 
 def adagrad(loss_or_grads, params, learning_rate=1.0, epsilon=1e-6):
+    """Adagrad updates [1]
+
+    Scale learning rates by dividing with the square root of accumulated
+    squared gradients.
+
+
+    Parameters
+    ----------
+    loss_or_grads : symbolic expression or list of expressions
+        A scalar loss expression, or a list of gradient expressions
+    params : list of shared variables
+        The variables to generate update expressions for
+    learning_rate : float or symbolic scalar
+        The learning rate controlling the size of update steps
+    epsilon : float or symbolic scalar,
```

Comment:
There shouldn't be a comma here (same thing further below), but that's probably not really critical.